jupyter-extension/src/plugins/mount/mountDrive.ts

```diff
@@ -123,6 +138,48 @@ export class MountDrive implements Contents.IDrive {
     };
   }
 
+  _fetchNextPage(
+    previousResponse: Contents.IModel,
+    key: string,
+    url: string,
+    getOptions: PartialJSONObject,
+  ): void {
+    // Stop fetching pages if we recieve a page less than expected file count of a page
+    if (previousResponse?.content?.length < PAGINATION_NUMBER) {
+      return;
+    }
+
+    // Queuing a promise without returning it to await lets us start the process of fetching the next pages in the background while
+    // continuing to render the the first page.
+    (async () => {
+      const nextResponse: Contents.IModel = await new Promise(
+        (resolve, reject) => {
+          return this._get(url, {
+            ...getOptions,
+            pagination_marker: previousResponse.content.slice(-1)[0].file_uri,
+          })
+            .then((nextResponse) => resolve(nextResponse))
+            .catch((e) => reject(e));
+        },
+      );
+
+      // Check to make sure that the cache file path has not changed so that we stop updating the contents and requesting
+      // new pages
+      if (this._cache.key !== key) {
```

Comment:
Is there any fear that the user could quickly switch directories and then switch back before this check? So the cache would be cleared but the cache key would still be the same?

Reply:
Ok I think I solved this by change it from a check for the same key to a check for the time the original directory change happened. Whenever an actual change in directory happens that restarts that first page request `this._cache.now` is set to `Date.now()` which is checked after each request before adding to the contents.